### PR TITLE
chore: check config policies on 'det notebook set priority'

### PR DESCRIPTION
**Pull request by stoksc**
## Ticket
CM-562

## Description
Add a constraint check to a stray priority change API.

## Test Plan
TBD.

[DET-123]: https://hpe-aiatscale.atlassian.net/browse/DET-123?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

**Review by kkunapuli — APPROVED** [review]

File: master/internal/command/command.go [hunk 1]
```diff
@@ -278,6 +279,16 @@ func (c *Command) garbageCollect() {
 }
 
 func (c *Command) setNTSCPriority(priority int, forward bool) error {
+	if smallerHigher, err := c.rm.SmallerValueIsHigherPriority(); err == nil {
+		ok, err := configpolicy.PriorityAllowed(int(c.Metadata.WorkspaceID), model.NTSCType, priority, smallerHigher)
```
Comment (amandavialva01) [inline master/internal/command/command.go:283]: I think we need to be calling 
```suggestion
		ok, err := configpolicy.PriorityUpdateAllowed(int(c.Metadata.WorkspaceID), model.NTSCType, priority, smallerHigher)
```
That function is implemented in the only [commit](https://github.com/determined-ai/determined/pull/10025/files) on main that this branch doesn't have lol 😆. `PriorityAllowed` checks against constraints but we also want to check that this field isn't set in an invariant config, which is done in `findAllowedPriority`, a helper in `PriorityUpdateAllowed`!

**Review by amandavialva01 — CHANGES_REQUESTED** [review]
This looks great! I think we need to rebase this branch onto main and change the helper being used so that we check for invariant configs as well!

File: master/internal/command/command.go (hunk 1 above)
Comment (kkunapuli) [reply master/internal/command/command.go:283]: I changed the name of the function (by request) in a recent PR. Sorry to change the code out from under you.

Comment (stoksc) [reply master/internal/command/command.go]: I've changed it. I didn't have any conflicts from git (merge button is still green), it was a semantic conflict. So thanks for the catch.

**Review by amandavialva01 — APPROVED** [review]
LGTM, awesome work!

File: master/internal/command/command.go (hunk 1 above)
Comment (amandavialva01) [reply master/internal/command/command.go]: Awesome! Change LGTM, approved!

**Pull request merged**